### ISSUE TITLE
Ability to match arrow function properties

### ISSUE THREAD

**Issue by suchipi** [bryanrsmith/eslint-plugin-sort-class-members#19]
It's common practice to use arrow functions as class properties instead of methods when you want the "method" to be bound to the instance:
```
class Foo {
  someBoundMethod = () => {
    // etc
  }
}
```
I want to treat these like methods, but treat non-function class properties differently.

It doesn't appear that it is currently possible to make this distinction. It'd be nice if I could do something like
```
{
  type: "property",
  propertyType: "ArrowFunctionExpression"
}
```
ie. `propertyType` refers to the node type of the `ClassProperty`'s `value`.

**Comment by bryanrsmith** [comment]
Thanks, @suchipi. Good suggestion. I do that often, too.

It shouldn't be too difficult to implement your suggestion of exposing a property type. Do you think it's sufficient, though?

> I want to treat these like methods

That is not what the `propertyType` change would do. That would let you select properties whose value matched a certain type, but they would still have to be placed before or after the methods group. They couldn't me mixed together. If you want to treat both function properties and methods the same and allow them to be intermixed we'd need a way to add a group that didn't care about the order of its sub-types. I've considered that in the past, but no one has ever asked for it.

**Comment by suchipi** [comment]
I was actually okay not mixing the two order-wise, though I think that feature could have value. Both features could be implemented independently of each other though. But, were you thinking of changing the default method group to allow mixed arrow function expression properties? In that case I think it'd make sense to implement them together at the same time.

**Comment by bryanrsmith** [comment]
Sounds good. I won't change the defaults. I'll try to look into adding
support for inspecting the property type this weekend. Mixed groups can
come later.

**Comment by bryanrsmith** [comment]
Published version 1.1.0 with this change. The API is as you suggest. Please let me know how it works for you!

Thanks,
Bryan
